### PR TITLE
[Elastic Agent] Cloud container legacy apm files.

### PR DESCRIPTION
**Pull request by simitt**
## What does this PR do?
Sync legacy apm `ingest` folder to `HOME` directory.

## Why is it important?
The APM Server fails to connect to Elasticsearch if it can't read the ingest pipeline definitions.

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/cmd/container.go [hunk 1]
```diff
@@ -532,26 +532,26 @@ func runLegacyAPMServer(streams *cli.IOStreams, path string) (*process.Info, err
...
+	if len(files) != 1 || !files[0].IsDir() {
+		return nil, errors.New("expected one directory")
 	}
-	f, err := filepath.Abs(files[0])
-	if err != nil {
-		return nil, errors.New(err, fmt.Sprintf("absPath for %s", files[0]))
+	apmDir := filepath.Join(path, files[0].Name())
+	// Extract the ingest pipeline definition to the HOME_DIR
+	if home := os.Getenv("HOME_PATH"); home != "" {
+		syncDir(filepath.Join(apmDir, "ingest"), filepath.Join(home, "ingest"))
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/agent/cmd/container.go:551]: You don't catch the error here? I think that should be handled so if syncing the directory fails its clear.

**Review by blakerouse — APPROVED** [review]
Looks good other than the missing error check.

File: x-pack/elastic-agent/pkg/agent/cmd/container.go [hunk 2]
```diff
@@ -532,26 +533,28 @@ func runLegacyAPMServer(streams *cli.IOStreams, path string) (*process.Info, err
...
-	files, err := filepath.Glob(pattern)
+	// Get the apm-server directory
+	files, err := ioutil.ReadDir(path)
 	if err != nil {
-		return nil, errors.New(err, fmt.Sprintf("searching apm-server in %s", pattern))
+		return nil, errors.New(err, fmt.Sprintf("reading directory %s", path))
 	}
-	if len(files) != 1 {
-		return nil, errors.New("multiple apm-server versions installed")
+	if len(files) != 1 || !files[0].IsDir() {
```
Comment (ruflin) [inline x-pack/elastic-agent/pkg/agent/cmd/container.go:546]: What happens if `len(files) == 0`?

Comment (simitt) [reply x-pack/elastic-agent/pkg/agent/cmd/container.go:546]: good point, will do a follow up. I honestly wish there was some basic setup with ES, Fleet Server and Kibana, to run some integration tests and not only the e2e testing living in another repository.

Comment (ruflin) [reply x-pack/elastic-agent/pkg/agent/cmd/container.go:546]: ++ on the integration tests.

Comment (simitt) [reply x-pack/elastic-agent/pkg/agent/cmd/container.go:546]: https://github.com/elastic/beats/issues/23821#issuecomment-814080040

Comment (simitt) [reply x-pack/elastic-agent/pkg/agent/cmd/container.go:546]: > What happens if len(files) == 0?

`if len(files) != 1` covers the case that it is `0`, in which case the error is returned; 
@ruflin am I missing something here, did you have something else in mind?

Comment (ruflin) [reply x-pack/elastic-agent/pkg/agent/cmd/container.go:546]: You are right, if it is 0 it only checks the first condition and also it never gets to line 549.

**Pull request merged**